crates/primitives/starknet/src/execution/contract_class_wrapper.rs

```diff
@@ -44,104 +47,191 @@ impl<'de> Deserialize<'de> for ProgramWrapper {
+                program: class.program.clone().into(),
+                entry_points_by_type: EntrypointMapV1Wrapper(unsafe {
+                    mem::transmute::<
+                        HashMap<EntryPointType, Vec<EntryPointV1>>,
+                        HashMap<EntryPointTypeWrapper, Vec<EntryPointV1Wrapper>>,
+                    >(class.entry_points_by_type.clone())
+                }),
+                hints: class.hints.clone().into(),
+            }),
         }
     }
 }
 
-impl MaxEncodedLen for ContractClassWrapper {
-    fn max_encoded_len() -> usize {
-        // This is the maximum size of a contract in starknet. https://docs.starknet.io/documentation/starknet_versions/limits_and_triggers/
-        Self::MAX_CONTRACT_BYTE_SIZE
+/// Wrapper type for a [HashMap<String, EntryPoint>] object. (It's not really a wrapper it's a
+/// copy of the type but we implement the necessary traits.)
+#[derive(Clone, Debug, PartialEq, Eq, Default, Constructor, Serialize, Deserialize)]
+pub struct EntrypointMapV0Wrapper(pub HashMap<EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>>);
+
+/// SCALE trait.
+impl Encode for EntrypointMapV0Wrapper {
+    fn encode_to<T: Output + ?Sized>(&self, dest: &mut T) {
+        // Convert the EntrypointMapV0Wrapper to Vec<(EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>)> to
+        // be able to use the Encode trait from this type. We implemented it for
+        // EntryPointV0Wrapper, derived it for EntryPointTypeWrapper so we can use it for
+        // Vec<(EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>)>.
+        let val: Vec<(EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>)> = self.0.clone().into_iter().collect();
+        dest.write(&Encode::encode(&val));
+    }
+}
+/// SCALE trait.
+impl Decode for EntrypointMapV0Wrapper {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+        // Convert the EntrypointMapV0Wrapper to Vec<(EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>)> to
+        // be able to use the Decode trait from this type. We implemented it for
+        // EntryPointV0Wrapper, derived it for EntryPointTypeWrapper so we can use it for
+        // Vec<(EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>)>.
+        let val: Vec<(EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>)> =
+            Decode::decode(input).map_err(|_| Error::from("Can't get EntrypointMap from input buffer."))?;
+        Ok(EntrypointMapV0Wrapper(HashMap::from_iter(val.into_iter())))
     }
 }
 
-/// Wrapper type for a [HashMap<String, EntryPoint>] object. (It's not really a wrapper it's a
+/// SCALE trait.
+impl TypeInfo for EntrypointMapV0Wrapper {
+    type Identity = Self;
+
+    // The type info is saying that the EntryPointByType must be seen as an
+    // array of bytes.
+    fn type_info() -> Type {
+        Type::builder()
+            .path(Path::new("EntrypointMapV0Wrapper", module_path!()))
+            .composite(Fields::unnamed().field(|f| f.ty::<[u8]>().type_name("EntrypointMapV0")))
+    }
+}
```

Comment:
could you create an issue to create a macro `scale_hashmap!` ? 
We're using this same piece of code in 3 # places already.

Reply:
This code will also be removed but I will check in the implementation in the appropriate repo and will create the issue

Reply:
Awesome thanks!